### ISSUE TITLE
Ignore words with non-latin characters

### ISSUE THREAD

**Issue by alystair** [streetsidesoftware/vscode-spell-checker#458]
![image](https://user-images.githubusercontent.com/159109/78343334-b01d9a00-7568-11ea-9d83-dc1489a85469.png)

Obviously Spanish and other pure latin char languages would have to be ignored manually by user, but Russian and other languages that use non-latin characters should automatically be ignored unless we're using that language dictionary?

**Comment by johnml1135** [comment]
I was able to get around it with this:
```
  "cSpell.includeRegExpList": [
    "\b[a-zA-Z0-9.]+\b"
  ],
```

**Comment by Jason3S** [comment]
It is necessary to explicitly ignore character sets. By default, the spell checker checks all text.

It is possible to tell the spell checker to ignore a character set using the `ignoreRegExpList` or only include text that matches expressions in `includeRegExpList`.

The spell checker uses JavaScript's builtin regexp engine. To use Unicode matching the `u` flag needs to be added.

It is also necessary to specify `Script_Extensions=` when using script names. See: [Unicode character class escape: \p{...}, \P{...} - JavaScript | MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape). It is always best to try out expressions at [regex101: build, test, and debug regex](https://regex101.com/).

### Using directive within a document

```js
// cspell:ignoreRegExp /[\p{Script_Extensions=Cyrillic}]+/gu
```

<img width="757" alt="image" src="https://github.com/streetsidesoftware/vscode-spell-checker/assets/3740137/31a40e60-1a19-4983-840c-889897d1c4c8">

### VS Code Settings

**`.vscode/settings.json`**
```jsonc
  "cSpell.ignoreRegExpList": ["/[\\p{Script_Extensions=Cyrillic}]+/gu"]
```

```

### Using CSpell config

**`cspell.json`**
```jsonc
{
  "ignoreRegExpList": ["/[\\p{Script_Extensions=Cyrillic}]+/gu"]
}
```

**`cspell.config.yaml`**
```yaml
ignoreRegExpList": 
  - '/[\p{Script_Extensions=Cyrillic}]+/gu'
```


## List of Character sets

Useful reference: [Unicode Scripts](https://www.regular-expressions.info/unicode.html#Unicode-Scripts:~:text=Unicode%20Scripts,-The%20Unicode%20standard)

List
- `Common`
- `Arabic`
- `Armenian`
- `Bengali`
- `Bopomofo`
- `Braille`
- `Buhid`
- `Canadian_Aboriginal`
- `Cherokee`
- `Cyrillic`
- `Devanagari`
- `Ethiopic`
- `Georgian`
- `Greek`
- `Gujarati`
- `Gurmukhi`
- `Han`
- `Hangul`
- `Hanunoo`
- `Hebrew`
- `Hiragana`
- `Inherited`
- `Kannada`
- `Katakana`
- `Khmer`
- `Lao`
- `Latin`
- `Limbu`
- `Malayalam`
- `Mongolian`
- `Myanmar`
- `Ogham`
- `Oriya`
- `Runic`
- `Sinhala`
- `Syriac`
- `Tagalog`
- `Tagbanwa`
- `TaiLe`
- `Tamil`
- `Telugu`
- `Thaana`
- `Thai`
- `Tibetan`
- `Yi`
